packages/nuxt/src/app/composables/error.ts

```diff
@@ -13,7 +13,7 @@ export interface NuxtError<DataT = unknown> extends H3Error<DataT> {}
 
 /** @since 3.0.0 */
 export const showError = <DataT = unknown>(
-    error: string | Error | Partial<NuxtError<DataT>>
+    error: Parameters<typeof createH3Error<DataT>>[0]
```

Comment:
Could we align it rather than use `Parameters`? 🙏  I might be wrong but worry this might cause issues with either TS perf or non-hoisted deps (if h3 isn't installed in root level).

Also, it's useful to have `NuxtError` as a separate interface that can be extended.

We can add a type assertion that isn't exported that will fail the build if these two aren't identical, so we keep the safety of ensuring we don't diverge from h3.